jdaviz/configs/cubeviz/plugins/tests/test_parsers.py

```diff
@@ -40,6 +40,41 @@ def image_hdu_obj():
     return fits.HDUList([fits.PrimaryHDU(), flux_hdu, mask_hdu, uncert_hdu])
 
 
+@pytest.fixture
+def image_hdu_obj_microns():
+    flux_hdu = fits.ImageHDU(np.ones((10, 10, 10)))
+    flux_hdu.name = 'FLUX'
+
+    mask_hdu = fits.ImageHDU(np.zeros((10, 10, 10)))
+    mask_hdu.name = 'MASK'
+
+    uncert_hdu = fits.ImageHDU(np.ones((10, 10, 10)))
+    uncert_hdu.name = 'ERR'
+
+    wcs = WCS(header={
+        'WCSAXES': 3, 'CRPIX1': 38.0, 'CRPIX2': 38.0, 'CRPIX3': 1.0,
+        'PC1_1 ': -0.000138889, 'PC2_2 ': 0.000138889,
+        'PC3_3 ': 8.33903304339E-11, 'CDELT1': 1.0, 'CDELT2': 1.0,
+        'CDELT3': 1.0, 'CUNIT1': 'deg', 'CUNIT2': 'deg', 'CUNIT3': 'um',
+        'CTYPE1': 'RA---TAN', 'CTYPE2': 'DEC--TAN', 'CTYPE3': 'WAVE-LOG',
+        'CRVAL1': 205.4384, 'CRVAL2': 27.004754, 'CRVAL3': 3.62159598486E-07,
+        'LONPOLE': 180.0, 'LATPOLE': 27.004754, 'MJDREFI': 0.0,
+        'MJDREFF': 0.0, 'DATE-OBS': '2014-03-30',
+        'RADESYS': 'FK5', 'EQUINOX': 2000.0
+    })
+
+    fits_primary = fits.PrimaryHDU()
+    fits_primary.header['CUNIT3'] = 'um'
+
+    flux_hdu.header.update(wcs.to_header())
+    flux_hdu.header['BUNIT'] = '1E-17 erg*s^-1*cm^-2*Angstrom^-1*pix^-1'
+
+    mask_hdu.header.update(wcs.to_header())
+    uncert_hdu.header.update(wcs.to_header())
```

Comment:
Realistically speaking, pipeline never put WCS in uncert and mask extensions. But if you want it anyway, only call `to_header` once and reuse the output instead of calling it 3x.

Reply:
Are you talking about [these lines](https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/cubeviz/plugins/tests/test_parsers.py#L34-#L38)? That's in main. I can remove it for this new test though.